### DISCUSSION TITLE
rabbitmq management api timeout issue

### DISCUSSION THREAD

**Discussion by ssk199441** [rabbitmq/rabbitmq-server#5749] [Other]
I am using Rabbitmq version (3.8.14), and using it as 8 node cluster using kubenetes.
In my prod environment , i have nearly
connections 25000
queues  29000
exchanges 56000
consumer 50000
channels 25000
and also lot user,permissions, bindings

I am taking backup using the management API `curl -k -u  username:password https://localhost:9450/api/definitions ` but it get timed out after one minute and my code clearly indicates it timed out at rabbitmq

My application is ssl enabled(HTTPS) , when i used that option my pod not coming up, is there any other option that i need to give o configure this.

How to avoid time out after one minute , when taking backup using management API, also i posed question rabbitmq google groups but i did't get enough answers
https://groups.google.com/g/rabbitmq-users/c/2dibT7etGC4/m/u4NpVEb5AQAJ?utm_medium=email&utm_source=footer

I have referred to following page increased the timedout
https://www.rabbitmq.com/management.html

# For RabbitMQ 3.7.9 and later versions.
#
# Configures HTTP (non-encrypted) listener timeouts
management.tcp.idle_timeout       = 120000
management.tcp.inactivity_timeout = 120000
management.tcp.request_timeout    = 120000

# For RabbitMQ 3.7.9 and later versions.
#
# Configures HTTPS (TLS-enabled) listener timeouts
management.ssl.idle_timeout       = 120000
management.ssl.inactivity_timeout = 120000
management.ssl.request_timeout    = 120000

**Comment by ssk199441** [comment]
Code
`#!/usr/bin/python3

import httpx
import asyncio

async def main():
    async with httpx.AsyncClient(verify=False) as client:
        r = await client.get('https://username:password@localhost:9450/api/definitions')
        print(r.text)

asyncio.run(main())`

Error logs:

Traceback (most recent call last):
  File "backup.py", line 15, in <module>
    asyncio.run(main())
  File "/usr/lib/python3.8/asyncio/runners.py", line 44, in run
    return loop.run_until_complete(main)
  File "/usr/lib/python3.8/asyncio/base_events.py", line 616, in run_until_complete
    return future.result()
  File "backup.py", line 11, in main
    async with session.get(pokemon_url,  ssl=False) as resp:
  File "/home/ironfist/python-rmq_backup/venv/lib/python3.8/site-packages/aiohttp/client.py", line 1138, in __aenter__
    self._resp = await self._coro
  File "/home/ironfist/python-rmq_backup/venv/lib/python3.8/site-packages/aiohttp/client.py", line 559, in _request
    await resp.start(conn)
  File "/home/ironfist/python-rmq_backup/venv/lib/python3.8/site-packages/aiohttp/client_reqrep.py", line 898, in start
    message, payload = await protocol.read()  # type: ignore[union-attr]
  File "/home/ironfist/python-rmq_backup/venv/lib/python3.8/site-packages/aiohttp/streams.py", line 616, in read
    await self._waiter
aiohttp.client_exceptions.ServerDisconnectedError: Server disconnected

**Comment by michaelklishin** [comment]
The only thing we can suggest without server logs is that RabbitMQ 3.8 has reached end of life in July this year.

If this is an HTTP API response timeout, then you can try a higher value or `rabbitmqctl export_definitions` with a binary format which is more CPU efficient (the command which does have a `--timeout` of its own).

Producing a complete backup with tens of thousands of entities is going to be CPU intensive and
the cluster may already be near exhaustion of the CPU resources it has available to it.

In the commercial RabbitMQ edition there is a feature that continuously transfers definitions to a separate cluster. Unfortunately, the very first transfer will include everything and be similar
to `rabbitmqctl export_definitions --format=erlang [/path/to/definitions.terms` in terms efficiency.

**Comment by ssk199441** [comment]
Thank you for taking the time to help me

Ya i tried with rabbitmq export definitions and it worked without any issue but i need to use management api, since i am calling the api from different pod from rabbitmq pod.

Also i noticed the version i am using is outdated. we are going upgrade it as soon as possible.

Is there any to fix the timeout as mentioned in the document

https://www.rabbitmq.com/management.html

> For RabbitMQ 3.7.9 and later versions.
> Configures HTTP (non-encrypted) listener timeouts
> management.tcp.idle_timeout = 120000
> management.tcp.inactivity_timeout = 120000
> management.tcp.request_timeout = 120000
> 
> For RabbitMQ 3.7.9 and later versions.
> Configures HTTPS (TLS-enabled) listener timeouts
> management.ssl.idle_timeout = 120000
> management.ssl.inactivity_timeout = 120000
> management.ssl.request_timeout = 120000

**Reply by michaelklishin** [reply]
Timeouts are a function of node load and how much work is involved in rendering a definition file. Under high load producing a large definition file doesn't have a "fix" beyond provisioning more resources, bumping the timeouts and performing the export in the Erlang term format using CLI tools during a window of low activity.

**Comment by ssk199441** [comment]
Refer: https://www.rabbitmq.com/management.html
All values are in milliseconds. Their defaults vary:

management.tcp.inactivity_timeout has the default of 300 seconds
management.tcp.request_timeout has the default of 60 seconds
management.tcp.idle_timeout has the default of 5 seconds

Is there any chance to increase this default values
if i change it, is there any chance the request will wait until i get response
